api/server/router/container/container_routes.go

```diff
@@ -517,6 +518,11 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		hostConfig.KernelMemory = 0
 	}
 
+	if hostConfig != nil && runtime.GOOS == "linux" && versions.LessThan(version, "1.42") {
```

Comment:
version 1.42 introduces support for this, but doesn't make it required. you updated client to enforce a value, but an  alternative client (let's say: docker-py) could call this API with a non-set ConsoleSize
 
We could just guard this by `len(hostConfig.ConsoleSize) < 0`, so that this basically is "optional"

Reply:
Unfortunately, `ConsoleSize` is an array (fixed size), not a slice, so we can't set it to `nil` `[2]uint{0, 0}` is the default value for a `[2]uint`, so what this does is "reset the value as if it wasn't set" on older API version running on Linux.